drivers/spi/spi_shell.c

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 202 Tridonic GmbH & Co KG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT spi_4002c000
+
+
+#include <shell/shell.h>
+#include <stdlib.h>
+#include <drivers/spi.h>
+#include <string.h>
+#include <sys/util.h>
+#include <stdlib.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(SPI_shell, CONFIG_LOG_DEFAULT_LEVEL);
+
+#define SPI_DEVICE_PREFIX "SPI_"
+//#define GPIO_DEVICE_PREFIX "GPIO_"
+
+/* Maximum bytes we can write or read at once */
+#define MAX_SPI_BYTES	16
```

Comment:
That's short. I'd go for a default of 128bytes. That said: make is Kconfig configurable